src/adapter.ts

```diff
@@ -0,0 +1,340 @@
+import {
+  InternalServerErrorException,
+  Logger,
+  RequestMethod,
+  NestApplicationOptions,
+  VersioningType
+} from '@nestjs/common'
+import { RawBodyRequest, VersioningOptions, VersionValue, VERSION_NEUTRAL } from '@nestjs/common/interfaces'
+import { RouterMethodFactory } from '@nestjs/core/helpers/router-method-factory'
+import { isString, isUndefined } from '@nestjs/common/utils/shared.utils'
+import {
+  json as bodyParserJson,
+  Options,
+  OptionsJson,
+  OptionsUrlencoded,
+  urlencoded as bodyParserUrlencoded
+} from 'body-parser'
+import { cors } from '@tinyhttp/cors'
+import * as http from 'http'
+import * as https from 'https'
+import { App, Response, Request } from '@tinyhttp/app'
+import { AbstractHttpAdapter } from '@nestjs/core'
+import { default as sirv, Options as SirvOptions } from 'sirv'
+import { Duplex } from 'stream'
+
+export interface SirvNestOptions extends SirvOptions {
+  prefix?: string
+}
+
+export type VersionedRoute = <TRequest extends Record<string, any> = any, TResponse = any>(
+  req: TRequest,
+  res: TResponse,
+  next: () => void
+) => any
+
+export class TinyHttpAdapter extends AbstractHttpAdapter {
+  private readonly routerMethodFactory = new RouterMethodFactory()
+  private readonly openConnections = new Set<Duplex>()
+  constructor(instance?: any) {
+    super(instance ?? new App())
+  }
+
+  public reply(response: Response, body: any, statusCode?: number) {
```

Comment:
why use `public` when all class accessors are public by default? 🤔

Reply:
Just keeping the code convention same as that of express and fastify

Reply:
got it